app/services/sign_in/token_response_generator.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module SignIn
+  class TokenResponseGenerator
+    LOGGER_PREFIX = 'V0::SignInController'
+
+    attr_reader :grant_type, :code, :code_verifier, :client_assertion, :client_assertion_type, :assertion,
+                :subject_token, :subject_token_type, :actor_token, :actor_token_type, :client_id, :cookies
+
+    def initialize(valid_params:, cookies:)
+      @grant_type = valid_params[:grant_type]
+      @code = valid_params[:code]
+      @code_verifier = valid_params[:code_verifier]
+      @client_assertion = valid_params[:client_assertion]
+      @client_assertion_type = valid_params[:client_assertion_type]
+      @assertion = valid_params[:assertion]
+      @subject_token = valid_params[:subject_token]
+      @subject_token_type = valid_params[:subject_token_type]
+      @actor_token = valid_params[:actor_token]
+      @actor_token_type = valid_params[:actor_token_type]
+      @client_id = valid_params[:client_id]
+      @cookies = cookies
+    end
+
+    def perform
+      case grant_type
+      when Constants::Auth::AUTH_CODE_GRANT
+        generate_client_tokens
+      when Constants::Auth::JWT_BEARER_GRANT
+        generate_service_account_token
+      else
+        raise Errors::MalformedParamsError.new(message: 'Grant type is not valid')
+      end
+    end
+
+    private
+
+    def generate_client_tokens
+      validated_credential = CodeValidator.new(code:, code_verifier:, client_assertion:,
+                                               client_assertion_type:).perform
+      session_container = SessionCreator.new(validated_credential:).perform
+
+      sign_in_logger.info('token', session_container.access_token.to_s)
```

Comment:
I think we should move the 'token' log back to the SiS Controller (it will have less context since there won't be an unencoded access token), but have an additional log here with this info. I just want to preserve the log coming from the class calling it, and it still seems important to have a log at the controller level itself (but it still might be useful to have the payload info from token, so could be in a log that's like `[TokenResponseGenerator] session created {context}`, and `[TokenResponseGenerator] generated service account token {context}`

Reply:
Yeah that makes sense. Do we want to log any context in the controller or just 
`[SignInService] [V0::SignInController] token`?

Reply:
I don't think there needs to be any context there unless you can think of any that's available at that level. Really it's just to preserve the log format with the rest of the routes and if we wanted the context we can still find it, just with a different class name